WordPress/Classes/Services/ReaderPostService.m

```diff
@@ -168,6 +168,35 @@ - (void)fetchPost:(NSUInteger)postID forSite:(NSUInteger)siteID isFeed:(BOOL)isF
     }];
 }
 
+- (void)fetchPostAtURL:(NSURL *)postURL
+               success:(void (^)(ReaderPost *post))success
+               failure:(void (^)(NSError *error))failure
+{
+    ReaderPostServiceRemote *remoteService = [[ReaderPostServiceRemote alloc] initWithWordPressComRestApi:[self apiForRequest]];
+    [remoteService fetchPostAtURL:postURL
+                          success:^(RemoteReaderPost *remotePost) {
+        if (!success) {
+            return;
+        }
+
+        ReaderPost *post = [self createOrReplaceFromRemotePost:remotePost forTopic:nil];
+
+        NSError *error;
+        BOOL obtainedID = [self.managedObjectContext obtainPermanentIDsForObjects:@[post] error:&error];
+        if (!obtainedID) {
+            DDLogError(@"Error obtaining a permanent ID for post. %@, %@", post, error);
```

Comment:
Would it make sense to call `failure()` with this error?
It looks like it could create and infinite `loading` state if this error happens, but I couldn't make it happen on tests.